File: ShopService/ShopService.Core/Entities/Order.cs
using Shared.Enums;

namespace ShopService.Core.Entities
{
	/// <summary>
	/// Order.
	/// </summary>
	public class Order : BaseEntity
	{
		/// <summary>
		/// Full price of order.
		/// </summary>
		public decimal FullPrice { get; set; }

		/// <summary>
		/// Basket.
		/// </summary>
		public Basket? Basket { get; set; }

		/// <summary>
		/// Basket id.
		/// </summary>
		public Guid BasketId { get; set; }

		public DeliveryMethod? DeliveryMethod { get; set; }

		public Guid DeliveryMethodId { get; set; }

		public string DeliveryExtras { get; set; } = string.Empty;

		public PaymentMethod? PaymentMethod { get; set; }

		public Guid PaymentMethodId { get; set; }

		public PaymentStatus PaymentStatus { get; set; }

		public DeliveryStatus DeliveryStatus { get; set; }

		public OrderStatus OrderStatus { get; set; }
	}
}

File: AdministrativeFront/src/app/contracts/order.ts
import { Good } from "./catalog";
import { DeliveryMethod, PaymentMethod } from "./methods";
import { ShopUser } from "./user";

export interface Order {
    id: string;
    basket: Basket;
    basketId: string;
    deliveryMethod: DeliveryMethod;
    paymentMethod: PaymentMethod;
    paymentStatus: PaymentStatus;
    deliveryStatus: DeliveryStatus;
    orderStatus: OrderStatus;
    fullPrice: number;
    createdAt: Date;
    updatedAt: Date;
    deliveryExtras: string;
}

export interface Basket {
    id: string;
    user: ShopUser;
    goods: BasketItem[]
}

export interface BasketItem {
    id: string;
    price: number;
    count: number;
    goodId: string;
    good: Good
}

export enum DeliveryStatus {
    Created,
    InDelivery,
    Delivered,
    Fail
}

export enum PaymentStatus {
    Created,
    Payed,
    Fail
}

export enum OrderStatus{
    Created,
    Payment,
    Delivery,
    Completed,
    Canceled
}